weave/trace/async_job_queue.py

```diff
@@ -85,9 +85,9 @@ def callback(f: Future[T]) -> None:
                 self._active_jobs.remove(f)
             exception = f.exception()
             if exception:
-                logger.error(f"Job failed with exception: {exception}")
                 if get_raise_on_captured_errors():
                     raise
+                logger.error(f"Job failed with exception: {exception}")
```

Comment:
Why does order here matter?

Reply:
if `get_raise_on_captured_errors` is set (something only done for testing), then we don't want the logger to log (this is true for all cases of this pattern) We effectively get the error bubbled up as a proper exception